AsyncDisplayKit/Details/ASRangeController.mm

```diff
@@ -168,7 +175,9 @@ - (void)_updateVisibleNodeIndexPaths
   NSSet<NSIndexPath *> *allCurrentIndexPaths = [[allIndexPaths set] copy];
   [allIndexPaths unionSet:_allPreviousIndexPaths];
   _allPreviousIndexPaths = allCurrentIndexPaths;
-  _currentRangeMode = rangeMode;
+  
+  // Update the current range mode based on interface state
+  _currentRangeMode = [ASRangeController rangeModeForInterfaceState:selfInterfaceState currentRangeMode:_currentRangeMode];
```

Comment:
I realized a flaw with my reasoning, which may require that you bring back the Boolean flag you had earlier. Specifically, although we do want this method to run at least once when the user forces a specific mode, we must also consider that the next time this method runs we may have some relevant new information like a scroll direction. It is important that this information be taken into account on that pass; and that information may not be available at this point, because this occurs potentially much earlier, even before the next update is scheduled...and the thing that schedules the next update may be a scroll event for example.

So, I think it is ultimately important that we query for the automatic range mode at the beginning of the update pass, but actually just skip the query step for the first range update after a mode is manually set.  I apologize for not realizing this when we were discussing earlier today, and asking you to remove the Boolean flag.